src/Centreon/Application/Controller/PlatformTopologyController.php

```diff
@@ -0,0 +1,126 @@
+     */
+    private $platformTopologyService;
+
+    /**
+     * PlatformTopologyController constructor
+     * @param PlatformTopologyServiceInterface $platformTopologyService
+     */
+    public function __construct(PlatformTopologyServiceInterface $platformTopologyService)
+    {
+        $this->platformTopologyService = $platformTopologyService;
+    }
+
+    /**
+     * Validate platform topology data according to json schema
+     *
+     * @param array<mixed> $platformToAdd data sent in json
+     * @param string $schemaPath
+     * @return void
+     * @throws PlatformTopologyException
+     */
+    private function validatePlatformTopologySchema(array $platformToAdd, string $schemaPath): void
+    {
+        $platformTopologySchemaToValidate = Validator::arrayToObjectRecursive($platformToAdd);
+        $validator = new Validator();
+        $validator->validate(
+            $platformTopologySchemaToValidate,
+            (object) ['ref' => 'file://' . $schemaPath],
+            Constraint::CHECK_MODE_VALIDATE_SCHEMA
+        );
+
+        if (!$validator->isValid()) {
+            $message = '';
+            foreach ($validator->getErrors() as $error) {
+                $message .= sprintf("[%s] %s\n", $error['property'], $error['message']);
+            }
+            throw new PlatformTopologyException($message);
+        }
+    }
+
+    /**
+     * Entry point to register a new server
+     *
+     * @param Request $request
+     * @return View
+     * @throws PlatformTopologyException
+     */
+    public function addPlatformToTopology(Request $request): View
+    {
+        // check user rights
+        $this->denyAccessUnlessGrantedForApiConfiguration();
+
+        // get http request content
+        $platformToAdd = json_decode((string) $request->getContent(), true);
+        if (!is_array($platformToAdd)) {
+            return $this->view([
+                'code' => Response::HTTP_BAD_REQUEST,
+                'message' => sprintf(
+                    _('Error when decoding sent data')
+                )
+            ]);
+        }
+
+        // validate data consistency
+        $this->validatePlatformTopologySchema(
+            $platformToAdd,
+            $this->getParameter('centreon_path')
+            . 'config/json_validator/latest/Centreon/PlatformTopology/AddServer.json'
+        );
+
+        try {
+            $setPlatformTopology = (new PlatformTopology())
+                ->setServerName($platformToAdd['server_name'])
+                ->setServerAddress($platformToAdd['address'])
+                ->setServerType($platformToAdd['server_type'])
+                ->setServerParentAddress($platformToAdd['server_parent']);
+
+            $this->platformTopologyService->addPlatformToTopology($setPlatformTopology);
+
+            return $this->view(null, Response::HTTP_CREATED);
+        } catch (\Throwable $ex) {
+            throw new PlatformTopologyException($ex->getMessage(), $ex->getCode(), $ex);
```

Comment:
```suggestion
            throw new PlatformTopologyException($ex->getMessage(), 0, $ex);
```

Reply:
The value "0", will be the one set as return code. For the script to be able to get the query computation status, a specific error code like 400 is required